app/scripts/controllers/dashboard.coffee

```diff
@@ -326,8 +326,15 @@ angular.module('edudashAppCtrl').controller 'DashboardCtrl', [
           loadingSrv.containerLoad schoolsP, document.getElementById mapId
           schoolsP.then (schools) ->
             if $scope.selected? and $scope.viewMode == 'schools'
+              $scope.selectedCodeYear =
+                code: $scope.selected.CODE
+                year: $scope.year
               $scope.select $scope.selected.CODE
 
+        $scope.$watch 'selectedCodeYear', (selectedCodeYear) -> if selectedCodeYear?
+          (findSchool selectedCodeYear.code).then (school)->
+            $scope.selected = school
```

Comment:
damn. ok. I'll try to see if I can find out why.

On Fri, Aug 21, 2015 at 4:06 PM, Ernesto Soliani <notifications@github.com>
wrote:

> In app/scripts/controllers/dashboard.coffee
> <https://github.com/WorldBank-Transport/edudash/pull/144#discussion_r37672840>
> :
>
> >                $scope.select $scope.selected.CODE
> >
> > +        $scope.$watch 'selectedCodeYear', (selectedCodeYear) -> if selectedCodeYear?
> > +          (findSchool selectedCodeYear.code).then (school)->
> > +            $scope.selected = school
>
> yes, that is was the reason It works when we talk. my bad. We need to
> figure out how to implement it.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/WorldBank-Transport/edudash/pull/144/files#r37672840>.
>